scheduler/src/modules/transactions/transaction.service.ts

```diff
@@ -0,0 +1,112 @@
+import { Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Transaction } from './transaction.entity';
+import { TransactionDetail } from './transaction-detail.entity';
+import { Cron } from '@nestjs/schedule';
+import { LogAsyncMethodExecution } from 'src/common/decorator/log-async-method-execution.decorator';
+import { generate } from 'src/helper/generator.helper';
+import { Like, Repository } from 'typeorm';
+import { BcHoliday } from './holiday.entity';
+
+@Injectable()
+export class TransactionService {
+  private holidays: string[] = [];
+  constructor(
+    @InjectRepository(Transaction)
+    private readonly transactionRepository: Repository<Transaction>,
+    @InjectRepository(TransactionDetail)
+    private readonly cfsTransactionDetailRepo: Repository<TransactionDetail>,
+    @InjectRepository(BcHoliday)
+    private readonly holidayRepository: Repository<BcHoliday>,
+  ) {
+    void this.initializeHolidays();
+  }
+
+  private async initializeHolidays(): Promise<void> {
+    this.holidays = await this.getBcHolidays();
+  }
+
+  async getAllTransactions(): Promise<Transaction[]> {
```

Comment:
Is this method being invoked. If not, please remove. It does not have any where clause and hence pretty dangerous is it executes in PROD Db as its expected to have thousands of records.